cylc/flow/suite_files.py

```diff
@@ -449,15 +453,26 @@ def register(reg=None, source=None, redirect=False, rundir=None):
             "suite name cannot be an absolute path: %s" % reg)
 
     if source is not None:
-        if os.path.basename(source) == SuiteFiles.SUITE_RC:
+        if os.path.basename(source) == SuiteFiles.FLOW_FILE:
             source = os.path.dirname(source)
     else:
         source = os.getcwd()
 
-    # suite.rc must exist so we can detect accidentally reversed args.
+    # flow.cylc must exist so we can detect accidentally reversed args.
     source = os.path.abspath(source)
-    if not os.path.isfile(os.path.join(source, SuiteFiles.SUITE_RC)):
-        raise SuiteServiceFileError("no suite.rc in %s" % source)
+    flow_file_path = os.path.join(source, SuiteFiles.FLOW_FILE)
+    if not os.path.isfile(flow_file_path):
+        # If using deprecated suite.rc, symlink it into flow.cylc:
+        suite_rc_path = os.path.join(source, SuiteFiles.SUITE_RC)
+        if os.path.isfile(suite_rc_path):
+            os.symlink(suite_rc_path, flow_file_path)
+            LOG.warning(
+                f'The filename "{SuiteFiles.SUITE_RC}" is deprecated in favor '
+                f'of "{SuiteFiles.FLOW_FILE}". The latter has been '
+                f'automatically created as a symlink to the former.')
+        else:
+            raise SuiteServiceFileError(
+                f"no flow.cylc or suite.rc in {source}")
```

Comment:
What do people think of this method of "upgrading" existing `suite.rc` to `flow.cylc`?

Reply:
Seems legit, the symlink isn't strictly necessary but it's not a problem or overhead and saves us replicating this logic elsewhere (e.g. `cylc scan`) so fine by me.

Reply:
Seems fine to me too :+1: